lib/ResolverFactory.js

```diff
@@ -66,19 +66,19 @@ const UseFilePlugin = require("./UseFilePlugin");
 /**
  * @typedef {Object} ResolveOptions
  * @property {AliasOptionEntry[]} alias
- * @property {string[][]} aliasFields
+ * @property {Set<string | string[]>} aliasFields
  * @property {(function(ResolveRequest): boolean)} cachePredicate
  * @property {boolean} cacheWithContext
- * @property {string[]} descriptionFiles
+ * @property {Set<string>} descriptionFiles
  * @property {boolean} enforceExtension
- * @property {string[]} extensions
+ * @property {Set<string>} extensions
  * @property {FileSystem} fileSystem
  * @property {Object | false} unsafeCache
  * @property {boolean} symlinks
  * @property {Resolver=} resolver
- * @property {(string | string[])[]} modules
+ * @property {Set<string | string[]>} modules
```

Comment:
Arrays in Sets are a little bit weird, but since this is only for performance I think we can ignore that and accept duplication here. Arrays are very rare anyway.

Reply:
yep, I mentioned this in PR description